### PR TITLE
Fix typo in docs

### PR DESCRIPTION
**Pull request by svilupp**
As part of the JuliaCon hackathon, I have fixed some typos I noticed in the docs and ran a few of the files through Grammarly.

There were no changes to the active code (only docs).

I'm happy to withdraw this PR if it's not helpful.

### CODE REVIEW

File: src/optimise/train.jl [hunk 1]
```diff
@@ -87,7 +87,7 @@ Here `pars` is produced by calling [`Flux.params`](@ref) on your model.
 (Or just on the layers you want to train, like `train!(loss, params(model[1:end-2]), data, opt)`.)
 This is the "implicit" style of parameter handling.
 
-Then, this gradient is used by optimizer `opt` to update the paramters:
+Then, this gradient is used by optimizer `opt` to update the parameters:
```
Comment (ToucheSir) [inline src/optimise/train.jl:90]:
```suggestion
This gradient is then used by optimizer `opt` to update the parameters:
```

File: docs/src/training/training.md [hunk 2]
```diff
@@ -179,8 +179,8 @@ function my_custom_train!(loss, ps, data, opt)
     end
     # Insert whatever code you want here that needs training_loss, e.g. logging.
     # logging_callback(training_loss)
-    # Insert what ever code you want here that needs gradient.
-    # E.g. logging with TensorBoardLogger.jl as histogram so you can see if it is becoming huge.
+    # Insert whatever code you want here that needs gradient.
+    # E.g. logging with TensorBoardLogger.jl as a histogram so you can see if it is becoming huge.
```
Comment (ToucheSir) [inline docs/src/training/training.md:183]:
```suggestion
    # Insert whatever code you want here that needs gradients.
    # e.g. logging histograms with TensorBoardLogger.jl to check for exploding gradients.
```

File: docs/src/training/training.md [hunk 3]
```diff
@@ -159,8 +159,8 @@ end
 ## Custom Training loops
 
 The `Flux.train!` function can be very convenient, especially for simple problems.
-Its also very flexible with the use of callbacks.
-But for some problems its much cleaner to write your own custom training loop.
+It's also very flexible with the use of callbacks.
+But for some problems it's much cleaner to write your own custom training loop.
```
Comment (ToucheSir) [inline docs/src/training/training.md:163]:
```suggestion
For some problems, however, it's much cleaner to write your own custom training loop.
```
I don't agree with that first line :smile:

File: docs/src/training/training.md [hunk 4]
```diff
@@ -93,7 +93,7 @@ using IterTools: ncycle
 data = ncycle([(x, y)], 3)
 ```
 
-It's common to load the `x`s and `y`s separately. In this case you can use `zip`:
+It's common to load the `x`s and `y`s separately. In this case, you can use `zip`:
```
Comment (ToucheSir) [inline docs/src/training/training.md:96]:
```suggestion
It's common to load the `x`s and `y`s separately. Here you can use `zip`:
```

File: docs/src/training/training.md [hunk 5]
```diff
@@ -57,18 +57,18 @@ ps = Flux.params(m)
...
 
 ## Model parameters
 
 The model to be trained must have a set of tracked parameters that are used to calculate the gradients of the objective function. In the [basics](../models/basics.md) section it is explained how to create models with such parameters. The second argument of the function `Flux.train!` must be an object containing those parameters, which can be obtained from a model `m` as `Flux.params(m)`.
 
 Such an object contains a reference to the model's parameters, not a copy, such that after their training, the model behaves according to their updated values.
 
-Handling all the parameters on a layer by layer basis is explained in the [Layer Helpers](../models/basics.md) section. Also, for freezing model parameters, see the [Advanced Usage Guide](../models/advanced.md).
+Handling all the parameters on a layer-by-layer basis is explained in the [Layer Helpers](../models/basics.md) section. Also, for freezing model parameters, see the [Advanced Usage Guide](../models/advanced.md).
```
Comment (ToucheSir) [inline docs/src/training/training.md:71]:
```suggestion
Handling all the parameters on a layer-by-layer basis is explained in the [Layer Helpers](../models/basics.md) section. For freezing model parameters, see the [Advanced Usage Guide](../models/advanced.md).
```

File: docs/src/training/optimisers.md [hunk 6]
```diff
@@ -135,7 +135,7 @@ end
 loss(rand(10)) # around 0.9
 ```
 
-In this manner it is possible to compose optimisers for some added flexibility.
+In this manner, it is possible to compose optimisers for some added flexibility.
```
Comment (ToucheSir) [inline docs/src/training/optimisers.md:138]:
```suggestion
It is possible to compose optimisers for some added flexibility.
```

File: docs/src/training/optimisers.md [hunk 7]
```diff
@@ -71,7 +71,7 @@ AdaBelief
 
 Flux's optimisers are built around a `struct` that holds all the optimiser parameters along with a definition of how to apply the update rule associated with it. We do this via the `apply!` function which takes the optimiser as the first argument followed by the parameter and its corresponding gradient.
 
-In this manner Flux also allows one to create custom optimisers to be used seamlessly. Let's work this with a simple example.
+In this manner Flux also allows one to create custom optimisers to be used seamlessly. Let's work on this with a simple example.
```
Comment (ToucheSir) [inline docs/src/training/optimisers.md:74]:
```suggestion
Flux also allows one to create custom optimisers. Let's work on this with a simple example.
```

**Review by ToucheSir — COMMENTED** [review]
Thanks! This is something I and others have wanted to do for a while but left as low priority, so it's great that you tackled it. I think the docs copy in general could use some wordsmithing, but that need not be addressed in this PR.

I've just taken the opportunity to touch up some existing grammatical awkwardness in the changed docs. If everything jives with you, we can merge.

**Comment by svilupp** [comment]
Yes, absolutely - it looks much better!

I didn’t dare to change it too much, because different people have different styles and I didn’t want to step on any toes.

**Pull request merged**